lib/_http_server.js

```diff
@@ -330,10 +330,11 @@ function Server(options, requestListener) {
   this[kIncomingMessage] = options.IncomingMessage || IncomingMessage;
   this[kServerResponse] = options.ServerResponse || ServerResponse;
 
-  const maxHeaderSize = options.maxHeaderSize;
-  if (maxHeaderSize !== undefined)
-    validateInteger(maxHeaderSize, 'maxHeaderSize', 0);
-  this.maxHeaderSize = maxHeaderSize;
+  validateInteger(
+    options.maxHeaderSize,
+    'maxHeaderSize',
+    { min: 0, allowUndefined: true });
+  this.maxHeaderSize = options.maxHeaderSize;
```

Comment:
unrelated change?

Reply:
It's related. The validator signature changes